crates/router/src/core/utils.rs

```diff
@@ -86,33 +86,31 @@ pub async fn construct_payout_router_data<'a, F>(
 
     let billing = payout_data.billing_address.to_owned();
 
-    let address = PaymentAddress {
-        shipping: None,
-        billing: billing.map(|a| {
-            let phone_details = api_models::payments::PhoneDetails {
-                number: a.phone_number.clone().map(Encryptable::into_inner),
-                country_code: a.country_code.to_owned(),
-            };
-            let address_details = api_models::payments::AddressDetails {
-                city: a.city.to_owned(),
-                country: a.country.to_owned(),
-                line1: a.line1.clone().map(Encryptable::into_inner),
-                line2: a.line2.clone().map(Encryptable::into_inner),
-                line3: a.line3.clone().map(Encryptable::into_inner),
-                zip: a.zip.clone().map(Encryptable::into_inner),
-                first_name: a.first_name.clone().map(Encryptable::into_inner),
-                last_name: a.last_name.clone().map(Encryptable::into_inner),
-                state: a.state.map(Encryptable::into_inner),
-            };
-
-            api_models::payments::Address {
-                phone: Some(phone_details),
-                address: Some(address_details),
-                email: a.email.to_owned().map(Email::from),
-            }
-        }),
-        payment_method_billing: None,
-    };
+    let billing_address = billing.map(|a| {
+        let phone_details = api_models::payments::PhoneDetails {
+            number: a.phone_number.clone().map(Encryptable::into_inner),
+            country_code: a.country_code.to_owned(),
+        };
+        let address_details = api_models::payments::AddressDetails {
+            city: a.city.to_owned(),
+            country: a.country.to_owned(),
+            line1: a.line1.clone().map(Encryptable::into_inner),
+            line2: a.line2.clone().map(Encryptable::into_inner),
+            line3: a.line3.clone().map(Encryptable::into_inner),
+            zip: a.zip.clone().map(Encryptable::into_inner),
+            first_name: a.first_name.clone().map(Encryptable::into_inner),
+            last_name: a.last_name.clone().map(Encryptable::into_inner),
+            state: a.state.map(Encryptable::into_inner),
+        };
+
+        api_models::payments::Address {
+            phone: Some(phone_details),
+            address: Some(address_details),
+            email: a.email.to_owned().map(Email::from),
+        }
+    });
+
+    let address = PaymentAddress::new(None, billing_address, None);
```

Comment:
These payment_method_billing changes would be required in payouts flow too right?

Reply:
Yes, wherever we were accessing address from payment_address directly, now they need to be accessed by getters, I have changed all the invocations to `get_payment_method_billing`

Reply:
FYI @kashif-m @Sakilmostak

Reply:
If you are talking about the ability to collect `payment_method_billing` in payouts request too, it will be taken up in a separate PR. These are the changes for payments only.